### PR TITLE
add additional test cases for sidecar with multiple ports in sidecar

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/model/sidecar_test.go [hunk 1]
```diff
@@ -1888,34 +2009,38 @@ func TestCreateSidecarScope(t *testing.T) {
...
-							found = true
-							break
 						}
+						break
 					}
 				}
-				if !found {
+				if !serviceFound {
 					t.Errorf("Expected service %v in SidecarScope but not found", s1.Hostname)
+				} else if len(ports) > 0 && !portsMatched {
```
Comment (ymesika) [inline pilot/pkg/model/sidecar_test.go:2030]: If I read the code correctly you don't actually need `ports`.
The if condition only checks `!portsMatched` and the error's got ports takes `s2.Ports`

Comment (Pawan-Bishnoi) [reply pilot/pkg/model/sidecar_test.go:2030]: I think it is required because s2 is accessible only inside the inner loop.

Comment (ramaraochavali) [reply pilot/pkg/model/sidecar_test.go:2030]: When ports are assigned, we break out of the loop and that is what we want to validate i.e. the service is matched and whether ports have been matced or not

File: pilot/pkg/model/service.go [hunk 2]
```diff
@@ -813,6 +817,14 @@ func (ports PortList) Equals(other PortList) bool {
 	return true
 }
 
+func (ports PortList) String() string {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/service.go:820]: We donot need such format for test

File: pilot/pkg/model/sidecar_test.go [hunk 3]
```diff
@@ -1888,34 +2009,38 @@ func TestCreateSidecarScope(t *testing.T) {
...
+					t.Errorf("Expected service %v found in SidecarScope but ports not merged correctly. want: %v, got: %v", s1.Hostname, s1.Ports, ports)
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/sidecar_test.go:2031]: %v can print ports. And maybe %+v

Comment (ramaraochavali) [reply pilot/pkg/model/sidecar_test.go:2031]: I tried both earlier and both printed this
```
Expected service proxy found in SidecarScope but ports not merged correctly. want: [0x14000f5adb0 0x1075d38e0], got: [0x1075d3820 0x1075d38a0 0x1075d38e0]
```

That is why I added that String() method

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**